algo.py

```diff
@@ -0,0 +1,33 @@
+from functools import wraps
+import time
+
+
+def measure_execution_time(function):
+    @wraps(function)
+    def wrapper(*args, **kwargs):
+        start_timestamp = time.perf_counter()
+        result = function(*args, **kwargs)
+        end_timestamp = time.perf_counter()
+        execution_time = end_timestamp - start_timestamp
+        print(f"Function {function.__name__}{args} took {execution_time}")
```

Comment:
if you accept both positional and keyword arguments, you probably need to include them both into your print